src/bucket/BucketApplicator.cpp

```diff
@@ -191,192 +191,92 @@ void
-             dd_sec, cu_sec, cd_sec, lu_sec, ld_sec, cdu_sec, cdd_sec, ccu_sec,
-             ccd_sec, csu_sec, eeu_sec, eed_sec, T_sec, total);
-    CLOG_DEBUG(Bucket,
-               "Apply-rates for {}-entry bucket {}.{} au:{} ad:{} tu:{} td:{} "
-               "ou:{} od:{} du:{} dd:{} cu:{} cd:{} lu:{} ld:{} "
-               "cdu:{} cdd:{} ccu:{} ccd:{} csu:{} eeu:{} eed:{} T:{}",
-               total, level, bucketName, au_sec, ad_sec, tu_sec, td_sec, ou_sec,
-               od_sec, du_sec, dd_sec, cu_sec, cd_sec, lu_sec, ld_sec, cdu_sec,
-               cdd_sec, ccu_sec, ccd_sec, csu_sec, eeu_sec, eed_sec, T_sec);
+    uint64_t T_sec, total;
+    CounterMap sec_counters;
+    getRates(now, sec_counters, T_sec, total);
+    CLOG_DEBUG(Bucket, "Apply-rates {} T:{}",
+               logStr(total, level, bucketName, sec_counters), T_sec);
 }
 
 void
 BucketApplicator::Counters::mark(BucketEntry const& e)
 {
     if (e.type() == LIVEENTRY || e.type() == INITENTRY)
     {
-        switch (e.liveEntry().data.type())
-        {
-        case ACCOUNT:
-            ++mAccountUpsert;
-            break;
-        case TRUSTLINE:
-            ++mTrustLineUpsert;
-            break;
-        case OFFER:
-            ++mOfferUpsert;
-            break;
-        case DATA:
-            ++mDataUpsert;
-            break;
-        case CLAIMABLE_BALANCE:
-            ++mClaimableBalanceUpsert;
-            break;
-        case LIQUIDITY_POOL:
-            ++mLiquidityPoolUpsert;
-            break;
-        case CONTRACT_DATA:
-            ++mContractDataUpsert;
-            break;
-        case CONTRACT_CODE:
-            ++mContractCodeUpsert;
-            break;
-        case CONFIG_SETTING:
-            ++mConfigSettingUpsert;
-            break;
-        case TTL:
-            ++mTTLUpsert;
-            break;
-        }
+        auto let = e.liveEntry().data.type();
+        auto iter = mCounters.find(let);
+        releaseAssert(iter != mCounters.end());
+        ++iter->second.first;
     }
     else
     {
-        switch (e.deadEntry().type())
-        {
-        case ACCOUNT:
-            ++mAccountDelete;
-            break;
-        case TRUSTLINE:
-            ++mTrustLineDelete;
-            break;
-        case OFFER:
-            ++mOfferDelete;
-            break;
-        case DATA:
-            ++mDataDelete;
-            break;
-        case CLAIMABLE_BALANCE:
-            ++mClaimableBalanceDelete;
-            break;
-        case LIQUIDITY_POOL:
-            ++mLiquidityPoolDelete;
-            break;
-        case CONTRACT_DATA:
-            ++mContractDataDelete;
-            break;
-        case CONTRACT_CODE:
-            ++mContractCodeDelete;
-            break;
-        case CONFIG_SETTING:
-            releaseAssert(false);
-            break;
-        case TTL:
-            ++mTTLDelete;
-            break;
-        }
+        auto let = e.deadEntry().type();
```

Comment:
We should preserve the invariant regarding deletion of CONFIG_SETTING (it's illegal, so we should crash)